markdown-collapsible.php

```diff
@@ -43,7 +50,7 @@ public function onMarkdownInitialized(Event $event)
 
                 $Block = array(
                     'name' => 'input',
-                    'markup' => '<input class="collapsible" id="'.$id.'" '.($name ? ' type="radio" name="'.$name.'"' : ' type="checkbox"').'><label class="collapsible" for="'.$id.'">'.$text.'</label><div class="collapsible">',
+                    'markup' => '<input class="collapsible" id="'.$id.'" '.($name ? ' type="radio" name="'.$name.'"' : ' type="checkbox"').'><label class="collapsible'.(!$do_autoscroll ? ' no-scroll' : '').'" for="'.$id.'"'. ($do_autoscroll ? 'style="scroll-margin:'.$offset.'px"' : '') .'>'.$text.'</label><div class="collapsible">',
```

Comment:
It's a good idea to use a class to disable the automatic scrolling. You can also do the same for the `scroll-margin`:

Add a `onAssetsInitialized` hook like this (pseudo code, not tested):
```php
public function onAssetsInitialized()
    {
        if ($this->config->get('plugins.markdown-collapsible.do_autoscroll'))
           $this->grav['assets']->addInlineCss(".soffset { scroll-margin: ".$this->config->get('scroll_offset').";}");
    }
```
That way, you don't have to add inline style to all label in the page only the class.

Reply:
Ahh, nice use of a platform method!

As I said, I don't really know Grav at all yet. This is definitely cleaner.

The `no-scroll` class on the label was also used to prevent the JS function from adding the scroll event listener though. I'll use the same `addInlineCss()` approach.

Reply:
hmmm, I can't find an API ref for anything that works like how you've described `addInlineCss()`. Unless I'm misunderstanding the docs, that feature of the Asset Manager is for injecting whole asset files or pipelines into the page; seems aimed at theme development.

I would bea really useful feature though. I wonder how hard it would be to PR Grav:dev too haha.

Anyway, unless you can point me at a doc I missed on how to do what you suggest, how do you want to solve this?

Reply:
See the new commit.

Tested quickly on my own simple site.

Reply:
LGTM, thanks!